# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: ipc/glue/UtilityProcessManager.h
# Commit: af562cec3d70
# Full Hash: af562cec3d706ee3ed605ffeade4d3ba982d2545
# Author: Ray Kraesig <rkraesig@mozilla.com>
# Date: 2023-10-27 04:08:13
# Regressor Bug: 1837079
# File Overlap Count: 2
# Description:
#   Bug 1837079 - [5/10] Create WinFileDialog IPDL protocol and implementation  r=gstoll,handyman,ipc-reviewers,nika,win-reviewers,mhowell
#   
#   Create and implement a new top-level IPC protocol, `PWinFileDialog`,
#   using the primitives from bug 1833450 and the new sandboxing type from
#   the previous commit.
# ==============================================================================

diff -r b5d52e5f75f2 -r af562cec3d70 ipc/glue/UtilityProcessManager.h
--- a/ipc/glue/UtilityProcessManager.h	Thu Oct 26 18:21:28 2023 +0000
+++ b/ipc/glue/UtilityProcessManager.h	Thu Oct 26 18:21:28 2023 +0000
@@ -23,6 +23,10 @@
 class WindowsUtilsParent;
 }  // namespace dom
 
+namespace widget::filedialog {
+class WinFileDialogParent;
+}  // namespace widget::filedialog
+
 namespace ipc {
 
 class UtilityProcessParent;
@@ -34,12 +38,19 @@
   friend class UtilityProcessParent;
 
  public:
+  template <typename T>
+  using Promise = MozPromise<T, nsresult, true>;
+
   using StartRemoteDecodingUtilityPromise =
-      MozPromise<Endpoint<PRemoteDecoderManagerChild>, nsresult, true>;
+      Promise<Endpoint<PRemoteDecoderManagerChild>>;
   using JSOraclePromise = GenericNonExclusivePromise;
 
-  using WindowsUtilsPromise =
-      MozPromise<RefPtr<dom::WindowsUtilsParent>, nsresult, true>;
+#ifdef XP_WIN
+  using WindowsUtilsPromise = Promise<RefPtr<dom::WindowsUtilsParent>>;
+
+  using WinFileDialogParent = widget::filedialog::WinFileDialogParent;
+  using WinFileDialogPromise = Promise<RefPtr<WinFileDialogParent>>;
+#endif
 
   static RefPtr<UtilityProcessManager> GetSingleton();
 
@@ -64,6 +75,10 @@
   // Releases the WindowsUtils actor so that it can be destroyed.
   // Subsequent attempts to use WindowsUtils will create a new process.
   void ReleaseWindowsUtils();
+
+  // Get a new Windows file-dialog utility-process actor. These are never
+  // reused; this will always return a fresh actor.
+  RefPtr<WinFileDialogPromise> CreateWinFileDialogAsync();
 #endif
 
   void OnProcessUnexpectedShutdown(UtilityProcessHost* aHost);